1205 download source

raman-bt opened the pull request.

kellykelly3 commented on KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/Bundle.properties-MERGED:
```diff
@@ -34,7 +34,7 @@ KeywordSearchIngestModule.startupMessage.failedToGetIndexSchema=Failed to get sc
 KeywordSearchResultFactory.query.exception.msg=Could not perform the query 
 OpenIDE-Module-Display-Category=Ingest Module
-OpenIDE-Module-Long-Description=Keyword Search ingest module.\n\nThe module indexes files found in the disk image at ingest time.\nIt then periodically runs the search on the indexed files using one or more keyword lists (containing pure words and/or regular expressions) and posts results.\n\nThe module also contains additional tools integrated in the main GUI, such as keyword list configuration, ...
+OpenIDE-Module-Long-Description=Keyword Search ingest module.\n\n\The module indexes files found in the disk image at ingest time.\n\It then periodically runs the search on the indexed files using one or more keyword lists (containing pure words and/or regular expressions) and posts results.\n\n\The module also contains additional tools integrated in the main GUI, such as keyword list configurati ...
```
We might want to talk to Richard about this one, I just changed this file to remove the \

rcordovano replied:
There is a problem with the Bundle.properties file for the Keyword Search NBM. I am going to fix it now.

raman-bt commented:
I don't know whats going on with these MERGED files.  I made no changes to
any of them intentionally.  All the changes here are what I got down from
develop branch.

Is you change merged to develop?



On Tue, Mar 5, 2019 at 4:07 PM kellykelly3 <notifications@github.com> wrote:

> *@kellykelly3* commented on this pull request.
> ------------------------------
>
> In
> KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/Bundle.properties-MERGED
> <https://github.com/sleuthkit/autopsy/pull/4591#discussion_r262690127>:
>
> > @@ -34,7 +34,7 @@ KeywordSearchIngestModule.startupMessage.failedToGetIndexSchema=Failed to get sc
>  KeywordSearchResultFactory.createNodeForKey.noResultsFound.text=No results found.
>  KeywordSearchResultFactory.query.exception.msg=Could not perform the query
>  OpenIDE-Module-Display-Category=Ingest Module
> -OpenIDE-Module-Long-Description=Keyword Search ingest module.\n\nThe module indexes files found in the disk image at ingest time.\nIt then periodically runs the search on the indexed files using one or more keyword lists (containing pure words and/or regular expressions) and posts results.\n\nThe module also contains additional tools integrated in the main GUI, such as keyword list configuration, keyword seach bar in the top-right corner, extracted text viewer and search results viewer showing highlighted keywords found.
> +OpenIDE-Module-Long-Description=Keyword Search ingest module.\n\n\The module indexes files found in the disk image at ingest time.\n\It then periodically runs the search on the indexed files using one or more keyword lists (containing pure words and/or regular expressions) and posts results.\n\n\The module also contains additional tools integrated in the main GUI, such as keyword list configuration, keyword seach bar in the top-right corner, extracted text viewer and search results viewer showing highlighted keywords found.
>
> We might want to talk to Richard about this one, I just changed this file
> to remove the \
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sleuthkit/autopsy/pull/4591#pullrequestreview-210919529>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEtFVJCZ4kDEvXdK6BpLB9IUHEcDzZEfks5vTtytgaJpZM4bfb4C>
> .
>

kellykelly3 commented on RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractSafari.java:
```diff
@@ -636,7 +641,19 @@ private BlackboardArtifact parseDownloadDictionary(Content dataSource, AbstractF
+        bbartifacts.add(bbart);
+        
+        // find the downloaded file and create a TSK_DOWNLOAD_SOURCE for it.
+        for (AbstractFile downloadedFile : fileManager.findFiles(dataSource, FilenameUtils.getName(path), FilenameUtils.getPath(path))) {
```
Is this the same for all the browsers?  Could you please add this to the createDownloadAttributes instead of this function?   Someday I would like to roll the createXXXAttributes out to all the extract subclasses.

kellykelly3 commented on RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Firefox.java:
```diff
@@ -532,6 +533,20 @@ private void getDownloadPreVersion24() {
+                    for (AbstractFile downloadedFile : fileManager.findFiles(dataSource, FilenameUtils.getName(downloadedFilePath), FilenameUtils.getPath(downloadedFilePath))) {
+                        BlackboardArtifact downloadSourceArt =  downloadedFile.newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_DOWNLOAD_SOURCE);
+                        downloadSourceArt.addAttribute(new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_URL,
+                            NbBundle.getMessage(this.getClass(), "Firefox.parentModuleName"), source)); //NON-NLS
```
Firefox.parentModelName should be changed to RecentActivityExtracterModuleFactory.getModuleName()

kellykelly3 commented on RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Firefox.java:
```diff
@@ -652,6 +668,20 @@ private void getDownloadVersion24() {
+                    for (AbstractFile downloadedFile : fileManager.findFiles(dataSource, FilenameUtils.getName(downloadedFilePath), FilenameUtils.getPath(downloadedFilePath))) {
+                        BlackboardArtifact downloadSourceArt =  downloadedFile.newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_DOWNLOAD_SOURCE);
+                        downloadSourceArt.addAttribute(new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_URL,
+                            NbBundle.getMessage(this.getClass(), "Firefox.parentModuleName"), url)); //NON-NLS
```
Change Firefox.parentModule Name to RecentActivityExtracterModuleFactory.getModuleName()

kellykelly3 commented on RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Chrome.java:
```diff
@@ -546,6 +548,20 @@ private void getDownload() {
+                    for (AbstractFile downloadedFile : fileManager.findFiles(dataSource, FilenameUtils.getName(fullPath), FilenameUtils.getPath(fullPath))) {
+                        BlackboardArtifact downloadSourceArt =  downloadedFile.newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_DOWNLOAD_SOURCE);
+                        downloadSourceArt.addAttribute(new BlackboardAttribute(ATTRIBUTE_TYPE.TSK_URL,
+                            NbBundle.getMessage(this.getClass(), "Chrome.parentModuleName"),
```
Change Chrome.parentModuleName to RecentActivityExtracterModuleFactory.getModuleName()

kellykelly3 commented on RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Chrome.java:
```diff
@@ -517,9 +518,10 @@ private void getDownload() {
             logger.log(Level.INFO, "{0}- Now getting downloads from {1} with {2}artifacts identified.", new Object[]{moduleName, temps, tempList.size()}); //NON-NLS
```
Missing a space before the word artifacts
